### ISSUE TITLE
launching firefox-esr crashes sway

### ISSUE THREAD

**Issue by bimmlerd** [swaywm/sway#8189]
### Please fill out the following:
- **Sway Version:** 1.9

- **Debug Log:** https://gist.github.com/bimmlerd/0cbb5d9f872ba6bfc5f9c14bfe839ee7

- **Configuration File:** default

- **Stack Trace:** https://gist.github.com/bimmlerd/0cbb5d9f872ba6bfc5f9c14bfe839ee7#file-bt-full-log (having trouble getting a good stacktrace - recompiling with debug symbols with debian results in missing core files 😓)

- **Description:**
  - start sway
  - start firefox-esr (via app launcher, might be bemenu
  - sway crashes before a window appears

**Comment by emersion** [comment]
Sounds like it's the same as this vmwgfx issue on the wlroots bug tracker: https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/3850

Nonetheless Sway shouldn't be crashing even if it fails to close the buffer handle.

**Comment by emersion** [comment]
Unfortunately, this backtrace isn't really usable...

**Comment by bimmlerd** [comment]
Yeah, I understand. I went off and tried random things myself, here's some datapoints which may or may not be helpful. If you have any suggestions on what I could try to help figure this out, I'm happy to oblige. (I am an engineer, instructions can be technical.) I've looked at backtraces for all of these crashes, they all look identical to the one I've posted, unfortunately - hence not including them again. FWIW these crashes are on `debian 6.10.4-arm64`, the rest of the system as mentioned in the first post.

1. sway doesn't always crash immediately - sometimes it takes up to ~5s until it crashes after launching `firefox` (without any further actions taken). If the delay happens, the log contains a great number of `_XSERVTransSocketUNIXAccept: accept() failed` lines. (Hence the zips - be aware that unpacking them will produce some 250MB of text)
1. Building sway with `-Db_sanitize=undefined` doesn't seem to change the behaviour. This is a log of such a run, which also encountered the "delayed" crash and contains the lines mentioned above: [sway-debug-log-sanitize-undefined.zip](https://github.com/user-attachments/files/16959587/sway-debug-log-sanitize-undefined.zip)
2. Building sway with `-Db_sanitize=thread` exhibits the same behaviour of crashing, but prints a stacktrace like the one we get from the coredump: 
[sway-debug-log-sanitize-thread.zip](https://github.com/user-attachments/files/16959594/sway-debug-log-sanitize-thread.zip)

3. Building sway with `-Db_sanitize=address,undefined` or `-Db_sanitize=address` seems to prevent the crash. At least, sway keeps running for at least a minute, after which I've killed it (via `killall sway` from a different tty, since I don't get a mouse when running from tty 🤷 ). Here's a log of such a run: 
[sway-debug-log-sanitize-address,undefined.zip](https://github.com/user-attachments/files/16959601/sway-debug-log-sanitize-address.undefined.zip)
 It's unclear to me whether not crashing is a random byproduct of ASAN or significant.

The next step forward to me seems like adding some instrumentation around the smoking gun of `Failed to close buffer handle: Invalid argument`, and maybe gleaning more info from that? Or is there a way to get a better backtrace?
